basiclibrary/src/test/java/basiclibrary/LibraryTest.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * This Java source file was generated by the Gradle 'init' task.
+ */
+package basiclibrary;
+
+import org.junit.Test;
+import static org.junit.Assert.*;
+
+public class LibraryTest {
+    @Test
+    public void testRoll() {
+        int[] arr = new int[4];
+        int length = arr.length;
+        assertEquals(
+                "This should check the length of the array.",
+                length,
+            Library.roll(4).length
+        );
+    }
+
+    @Test
+    public void testContainsDuplicates_false() {
+        int[] arr = new int[]{1,2,3};
+        assertEquals(
+                "This should return false as there are no duplicates in your array.",
+                false,
+                Library.containsDuplicates(arr)
+        );
+    }
+
+    @Test
+    public void testContainsDuplicates_true() {
+        int[] arr = new int[]{1,2,3,3};
+        assertEquals(
+                "This should return true as there are duplicates in your array.",
+                true,
+                Library.containsDuplicates(arr)
+        );
+    }
+
+    @Test
+    public void testAvgArray() {
+        int[] arr = new int[]{1,2,3,4};
+        double answer = 2.5;
+        //ERROR: Use assertEquals(expected, actual, delta) to compare floating-point numbers?
```

Comment:
yes, testing doubles is a little tricky, but your comment is correct. the delta provides what precision you are looking for